dvc/io/__init__.py

```diff
@@ -0,0 +1,152 @@
+import sys
+from collections import defaultdict
+from typing import Any, Dict, Iterable, Optional, TextIO
+
+from funcy import cached_property
+
+from dvc.progress import Tqdm
+from dvc.utils import colorize
+
+NEWLINE = "\n"
+SEP = " "
+
+
+class Formatter:
+    def __init__(self, theme: Dict = None, defaults: Dict = None) -> None:
+        theme = theme or {
+            "success": {"color": "green", "style": "bold"},
+            "warn": {"color": "yellow"},
+            "error": {"color": "red", "style": "bold"},
+        }
+        self.theme = defaultdict(lambda: defaults or {}, theme)
+
+    def format(self, message: str, style: str = None, **kwargs) -> str:
+        return colorize(message, **self.theme[style])
+
+
+class IO:
```

Comment:
I'm still in favor of naming this something besides `IO`, whether it's `UI` or `Console` or something else entirely.

`IO` to me still makes it look like this is supposed to be related to `io.IOBase` or `io.TextIOBase` for reading/writing text streams. We do have a `write` method, but it isn't really an arbitrary stream write, it's a formatted wrapper for `print`.